core/src/main/java/org/elasticsearch/tribe/TribeService.java

```diff
@@ -494,4 +573,28 @@ private void addNewIndex(ClusterState tribeState, ClusterBlocks.Builder blocks,
             }
         }
     }
+
+    private static ClusterService getClusterService(Node node) {
+        return node.injector().getInstance(ClusterService.class);
+    }
+
+    // pkg-private for testing
+    static Map<String, MetaData.Custom> mergeChangedCustomMetaData(Set<String> changedCustomMetaDataTypeSet,
+                                                                   Function<String, List<MetaData.Custom>> customMetaDataByTribeNode) {
```

Comment:
we can make this a function to `List<MergableCustomMetaData>` - we alreay check with instance of in the implementation of it.